bugsnag-android-core/src/main/java/com/bugsnag/android/SessionFilenameInfo.kt

```diff
@@ -27,29 +29,62 @@ internal data class SessionFilenameInfo(
          * Generates a filename for the session in the format
          * "[UUID][timestamp]_v2.json"
          */
-        fun toFilename(timestamp: Long, uuid: String): String {
-            return "${uuid}${timestamp}_v2.json"
+        fun toFilename(apiKey: String, timestamp: Long, uuid: String): String {
+            return "${apiKey}_${uuid}${timestamp}_v2.json"
         }
 
         @JvmStatic
-        fun defaultFilename(): String {
-            return toFilename(System.currentTimeMillis(), UUID.randomUUID().toString())
+        fun defaultFilename(
+            obj: Any,
+            config: ImmutableConfig
+        ): SessionFilenameInfo {
+            val sanitizedApiKey = when (obj) {
+                is Session -> obj.apiKey
+                else -> config.apiKey
+            }
+
+            return SessionFilenameInfo(
+                sanitizedApiKey,
+                System.currentTimeMillis(),
+                UUID.randomUUID().toString()
+            )
         }
 
-        fun fromFile(file: File): SessionFilenameInfo {
+        fun fromFile(file: File, config: ImmutableConfig): SessionFilenameInfo {
             return SessionFilenameInfo(
+                findApiKeyInFilename(file, config),
                 findTimestampInFilename(file),
                 findUuidInFilename(file)
             )
         }
 
         private fun findUuidInFilename(file: File): String {
-            return file.name.substring(0, uuidLength - 1)
+            val uuidWithTimestamp = file.name.substringAfter('_')
+            val uuidName = if (uuidWithTimestamp.length >= uuidLength) {
+                uuidWithTimestamp.take(uuidLength)
+            } else {
+                null
+            }
+            return uuidName.takeUnless { it.isNullOrBlank() } ?: ""
         }
 
         @JvmStatic
         fun findTimestampInFilename(file: File): Long {
-            return file.name.substring(uuidLength, file.name.indexOf("_")).toLongOrNull() ?: -1
+            return file.name.substringAfter('_')
+                .drop(uuidLength)
+                .substringBefore('_')
+                .toLongOrNull() ?: -1
+        }
+
+        fun findApiKeyInFilename(file: File, config: ImmutableConfig): String {
+            val apiKeyEnd = file.name.indexOf('_')
```

Comment:
As above, the '_' will pickup the "_v2.json" in existing filenames, returning apiKeys like "9df2622a-23b4-4e57-8962-a42152d0face1687165348810"